packages/scaffold-ui/src/views/w3m-connecting-farcaster-view/index.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  AccountController,
+  ConnectionController,
+  ConnectorController,
+  CoreHelperUtil,
+  ModalController,
+  RouterController,
+  SnackController,
+  StorageUtil,
+  ThemeController
+} from '@web3modal/core'
+import { customElement } from '@web3modal/ui'
+import { LitElement, html } from 'lit'
+import { state } from 'lit/decorators.js'
+import styles from './styles.js'
+
+@customElement('w3m-connecting-farcaster-view')
+export class W3mConnectingFarcasterView extends LitElement {
+  public static override styles = styles
+
+  // -- Members ------------------------------------------- //
+  private unsubscribe: (() => void)[] = []
+
+  // -- State & Properties -------------------------------- //
+  protected timeout?: ReturnType<typeof setTimeout> = undefined
+
+  @state() private socialProvider = AccountController.state.socialProvider
+
+  @state() protected uri = AccountController.state.farcasterUrl
+
+  @state() protected ready = false
+
+  @state() protected loading = false
+
+  public authConnector = ConnectorController.getAuthConnector()
+
+  public constructor() {
+    super()
+    this.unsubscribe.push(
+      ...[
+        AccountController.subscribe(val => {
```

Comment:
Can we use `subscribeKey` instead of `subscribe` to lower the re-renders. `subscribeKey` only triggered when the given key updated while `subscribe` triggered when any of the unused fields changed too.